src/baseline_bots/bots/smart_order_accepter_bot.py

```diff
@@ -421,6 +475,21 @@ def bad_move(self, order: str) -> bool:
 
         return False
 
+    def update_allies_and_foes(self) -> None:
+        power_stance = self.stance.stance[self.power_name]
+
+        self.allies = [
+            pow for pow in self.opponents if power_stance[pow] > self.ally_threshold
+        ]
+        self.foes = [
+            pow for pow in self.opponents if power_stance[pow] <= self.enemy_threshold
+        ]
+        self.neutral = [
+            pow
+            for pow in self.opponents
+            if self.enemy_threshold < power_stance[pow] <= self.ally_threshold
+        ]
```

Comment:
I'd feel better about this if there were an assert here to check that there was no overlap.

Reply:
ok added.